lib/systems/architectures.nix

```diff
@@ -28,6 +29,12 @@ rec {
     znver1         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx" "avx2"          "fma"        ];
     znver2         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx" "avx2"          "fma"        ];
     znver3         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx" "avx2"          "fma"        ];
+    # Microarchitecture levels
+    # https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels
+    x86-64         = [ ];
+    x86-64-v2      = [ "sse3" "ssse3" "sse4_1" "sse4_2"                                                  ];
```

Comment:
missing: `lahf-sahf`, `cmpxchg16b`, `popcnt` AFAIK?

do we know x86_64-v2 processors that **do not have** `aes-ni` ?

Reply:
At least the wiki pages says that `aes-ni` is not included in the feature levels  `Instruction set extensions not concerned with general-purpose computation, including [AES-NI](https://en.wikipedia.org/wiki/AES-NI) and [RDRAND](https://en.wikipedia.org/wiki/RDRAND), are excluded from the level requirements.` https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels at the very bottom of the chapter.

Reply:
i am not sure but about the other instruction lists, isnt that thing just used to check  `stdenv.hostPlatform.**Support`? If thats the case it would not give not much benefit to add those instructions if we do not check for their respective feature on any package. If we add a complete list of instructions (independent if we are using them for a feature check) we will also have to re-work all of the listed cpu tyes, as they are also missing those instructions then.
As far as i can see those feature list is not really passed to the compiler (in general), they get the march and mtune values and do then pick for the right instructions aint they?

Reply:
> At least the wiki pages says that `aes-ni` is not included in the feature levels `Instruction set extensions not concerned with general-purpose computation, including [AES-NI](https://en.wikipedia.org/wiki/AES-NI) and [RDRAND](https://en.wikipedia.org/wiki/RDRAND), are excluded from the level requirements.` https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels at the very bottom of the chapter.

That was not the point of my question: is there a x86_64-v2 processor that **do not have** `aes-ni` ? Not exactly the same as: "is `aes-ni` included in the theoretical microarchitecture level specification of `x86_64-v2` ?"

Reply:
Sorry I misunderstood you because of the context of the other question. if this [site](https://en.wikichip.org/wiki/intel/microarchitectures/nehalem_(client) ) is correct Nehalem does not support AES instructions.

Reply:
I am not planning to add new entries to those lists. That is a design decision I don't want to deal with right now. If you have feedback about the existing entries, especially if they are wrong, please let me know. :)

Reply:
On my personal opinion the list is okay for what it's used atm in the current design. The only thing that is missing in that PR here is to solve the bootstrapping issue. On the other PR i already posted my solution (https://github.com/NixOS/nixpkgs/commit/e8914d22114cfc9f042937f6e578cab4ed16dd62), that let me build a system without any other hacks. To be full the other added architecture (nahlem) should be added to. I do not know if that is useful input for the PR